### PR TITLE
nheko: 0.6.3 -> 0.6.4, mtxclient: 0.2.0 -> 0.2.1

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Update package nheko

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"` (Non depend on it)
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [X] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

#### Other information

- Updated the URL of mtxclient to use the `Nheko-Reborn` GitHub namspace.
- Made both nheko and mtxclient aware of nlohmann_json with `-Dnlohmann_json_DIR` in `cmakeFlags`.

### CODE REVIEW

File: pkgs/applications/networking/instant-messengers/nheko/default.nix [hunk 1]
```diff
@@ -1,8 +1,10 @@
 { lib, stdenv, fetchFromGitHub
 , cmake, cmark, lmdb, qt5, qtmacextras, mtxclient
-, boost, spdlog, olm, pkgconfig
+, boost, spdlog, olm, pkgconfig, nlohmann_json
 }:
 
+# These hashes and revisions are based on those from here:
+# https://github.com/Nheko-Reborn/nheko/blob/v0.6.4/deps/CMakeLists.txt#L52
```
Comment (matthiasbeyer) [inline pkgs/applications/networking/instant-messengers/nheko/default.nix]: Seems to be not relevant anymore, is it?

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Besides the one thing, this seems fine to me (not test-build this though).

**Comment by doronbehar** [comment]
Why not?
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

File: pkgs/applications/networking/instant-messengers/nheko/default.nix (hunk 1 above)
Comment (matthiasbeyer) [reply pkgs/applications/networking/instant-messengers/nheko/default.nix]: So it seems that the comment is there to tell the next person looking at the package that the hashes in `rev` are imported from that `CMakeLists` file...

I'd rather put this note in the commit message, which is (IMO, tbh) the first place to look at when curious about such a thing... not in the file itself at some other place...

Maybe someone else has a different opinion though...

Comment (doronbehar) [reply pkgs/applications/networking/instant-messengers/nheko/default.nix]: Although I'm a beginner Nix user, I did understand what these lines meant - I figured these are dependencies pulled manually by the Nix build system. Therefor, I figured that I may need to update these as well according to the new version's requirement. My first attempt was to search in nheko's source tree for any mention of what specific version does it pull. At first I thought these will be git submodules versions but then I've reached that `CMakeLists.txt` file and found these revisions' hashes.

My opinion is that when someone will want to update Nheko to 7.0.0, hopefully he will see this comment which importantly has a v0.6.4 URL, and will check if the revisions have had an update in 7.0.0.

Additionally, not every contributor's `git blame` skills are good enough that he will easily find the original commit message (not mine) that added these dependencies which have hopefully (I haven't checked that myself) mentioned where did the revisions came from.

**Comment by Ekleog** [comment]
Looks great, thanks! I agree with you that more comments are most often better, and the less people have to search for information the better.

Here I'd maybe have a doubt about this `0.6.4` in the URL, fearing it might get out-of-sync from the code below, but it's no big deal and there's no real better way of handling it I think, so… Thank you! :)

**Pull request merged**